Endpoint.js

```diff
@@ -0,0 +1,98 @@
+const { URL } = require('universal-url')
+const defaultFetch = require('nodeify-fetch')
+
+class Endpoint {
+  constructor ({ endpointUrl, factory, fetch, headers, password, storeUrl, updateUrl, user }) {
+    this.endpointUrl = endpointUrl
+    this.fetch = fetch || defaultFetch
```

Comment:
Or maybe it should be like this?

```suggestion
  constructor ({ endpointUrl, factory, fetch = defaultFetch, headers, password, storeUrl, updateUrl, user } = {}) {
    this.endpointUrl = endpointUrl
    this.fetch = fetch
```